pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -180,16 +180,24 @@ func convertRbacRulesToFilterConfig(service string, roles []model.Config, bindin
 		Policies: map[string]*policyproto.Policy{},
 	}
 
+	permissiveRbac := &policyproto.RBAC{
+		Action:   policyproto.RBAC_ALLOW,
+		Policies: map[string]*policyproto.Policy{},
+	}
+
 	for _, role := range roles {
-		log.Debugf("checking role %v for service %v", role.Name, service)
-		principals := convertToPrincipals(roleToBinding[role.Name])
+		enforcedPrincipals, permissivePrincipals := convertToPrincipals(roleToBinding[role.Name])
+		principals := make([]*policyproto.Principal, 0)
```

Comment:
Why don't you have the  convert function above the return the full list, based on role.Mode and binding mode?

Reply:
push the logic down to convertToPrincipals